### PR TITLE
Stop to ignore some CS rules

### PR DESCRIPTION
**Pull request by carusogabriel**
Following #251

### CODE REVIEW

File: lib/Doctrine/Common/Cache/PredisCache.php [hunk 1]
```diff
@@ -58,7 +58,7 @@ protected function doSaveMultiple(array $keysAndValues, $lifetime = 0)
             foreach ($keysAndValues as $key => $value) {
                 $response = (string) $this->client->setex($key, $lifetime, serialize($value));
```
Comment (carusogabriel) [inline lib/Doctrine/Common/Cache/PredisCache.php]: Even using `(string)` here, looks like it returns an object :(

File: lib/Doctrine/Common/Cache/LegacyMongoDBCache.php [hunk 2]
```diff
@@ -101,7 +101,11 @@ protected function doSave($id, $data, $lifeTime = 0)
             return false;
         }
 
-        return ($result['ok'] ?? 1) == 1;
+        if (! isset($result['ok'])) {
+            return true;
+        }
+
+        return $result['ok'] === 1.0;
```
Comment (Majkl578) [inline lib/Doctrine/Common/Cache/LegacyMongoDBCache.php]: You can't ever safely compare floats like this.

File: lib/Doctrine/Common/Cache/LegacyMongoDBCache.php [hunk 3]
```diff
@@ -111,7 +115,11 @@ protected function doDelete($id)
     {
         $result = $this->collection->remove(['_id' => $id]);
 
-        return ($result['ok'] ?? 1) == 1;
+        if (! isset($result['ok'])) {
+            return true;
+        }
+
+        return $result['ok'] === 1.0;
```
Comment (Majkl578) [inline lib/Doctrine/Common/Cache/LegacyMongoDBCache.php]: You can't ever safely compare floats like this.

File: lib/Doctrine/Common/Cache/LegacyMongoDBCache.php [hunk 4]
```diff
@@ -122,7 +130,11 @@ protected function doFlush()
         // Use remove() in lieu of drop() to maintain any collection indexes
         $result = $this->collection->remove();
 
-        return ($result['ok'] ?? 1) == 1;
+        if (! isset($result['ok'])) {
+            return true;
+        }
+
+        return $result['ok'] === 1.0;
```
Comment (Majkl578) [inline lib/Doctrine/Common/Cache/LegacyMongoDBCache.php]: You can't ever safely compare floats like this.

File: lib/Doctrine/Common/Cache/PredisCache.php [hunk 5]
```diff
@@ -58,7 +58,7 @@ protected function doSaveMultiple(array $keysAndValues, $lifetime = 0)
...
 
-                if ($response == 'OK') {
+                if ($response->payload === 'OK') {
```
Comment (Majkl578) [inline lib/Doctrine/Common/Cache/PredisCache.php]: Looks like functional change that needs a regresion test.

File: phpcs.xml.dist [hunk 6]
```diff
@@ -16,9 +16,6 @@
 
     <rule ref="Doctrine">
         <exclude name="SlevomatCodingStandard.TypeHints.DeclareStrictTypes"/>
-        <exclude name="SlevomatCodingStandard.TypeHints.TypeHintDeclaration.MissingParameterTypeHint"/>
-        <exclude name="SlevomatCodingStandard.TypeHints.TypeHintDeclaration.MissingReturnTypeHint"/>
```
Comment (Majkl578) [inline phpcs.xml.dist]: These two above must stay for 1.x.

**Comment by Majkl578** [comment]
@carusogabriel friendly ping

**Comment by carusogabriel** [comment]
@Majkl578 Feel free to finish it, I don't have the knowledge for :(

**Comment by carusogabriel** [comment]
Closing as I won't be able to finish it. Anyone is encourage to do it :blush:

**Pull request closed without merging**